src/app/providers/data-store/data-store.ts

```diff
@@ -36,6 +40,164 @@ export class DataStoreProvider {
+      for (const key in indexdbData) {
+        await this.storage.set(key, JSON.stringify(indexdbData[key]));
+      }
+    }
+  }
+
+  /**
+   * Retrieves and parses a value by key from the given IndexedDB object store.
+   *
+   * @param key - The key to look up in the object store.
+   * @param objectStore - The IndexedDB object store to query.
+   * @returns A promise resolving to the parsed JSON value, or null if not found.
+   */
+  async getIndexDBDataByKey(key: string, objectStore: IDBObjectStore): Promise<Record<string, unknown>> {
+    return new Promise((resolve, reject) => {
+      //try to get the value from the object store by key
+      const request = objectStore.get(key);
+      // Handle the success and error events
+      request.onsuccess = (event) => {
+        // If the value is found, parse it and resolve the promise
+        try {
+          // Get the result from the event
+          const value = get(event, 'target.result');
+          // If the value is null or undefined, resolve with null
+          if (!value) return resolve(null);
+          // Parse the value as JSON and resolve the promise
+          resolve(JSON.parse(value));
+        } catch (err) {
+          this.reportLog('getIndexDBDataByKey', '', err);
+          reject(err);
+        }
+      };
+      request.onerror = (event) => {
+        // If there is an error, log it and reject the promise
+        const error = get(event, 'target.error');
+        this.reportLog('getIndexDBDataByKey', '', error);
+        reject(error);
+      };
+    });
+  }
+
+  async getIndexDBData(): Promise<Record<string, unknown>> {
+    return new Promise((resolve, reject) => {
+      // Open the IndexedDB database
+      const request = window.indexedDB.open('_ionicstorage');
+      // Handle the success and error events
+      request.onsuccess = async (event) => {
+        // Get the database from the event
+        const db: IDBDatabase = get(event, 'target.result');
+        // If the database is not found, resolve with null
+        if (!db) {
+          return resolve(null);
+        }
+        // Get the names of all object stores in the database
+        const storeNames = Array.from(db.objectStoreNames);
+        // If there are no object stores, resolve with null
+        if (!storeNames.length) {
+          return resolve(null);
+        }
+        // Start a transaction to read from the object store
+        const transaction = db.transaction(storeNames, 'readonly');
+        // Get the object store from the transaction
+        const objectStore = transaction.objectStore('_ionickv');
+        // If the object store is not found, resolve with null
+        if (!objectStore) {
+          return resolve(null);
+        }
+        // Create an empty object to hold the data
+        const holdingJSON = {};
+        // Define the keys to be migrated
+        const keysToBeMigrated = [LocalStorageKey.JOURNAL, LocalStorageKey.EXAMINER_STATS_KEY];
```

Comment:
Is it worth bringing over TESTS too or is all of the data housed in the Journal key?

Reply:
agreed, TESTS would allow unuploaded tests to be saved.  

just for more context we've discussed config and the 3 token storage items and they are a definite no to migration